package/MDAnalysis/analysis/Aggregation_size.py

```diff
@@ -0,0 +1,370 @@
+will be considered as a criterion for considering two molecules as a one
+aggregate. Besides the cut_off distance, the two atoms' name (atom_1 and
+atom_2) should be defined. The default value for cut_off distance is 8.5
+Angstrom while there is no default value for atom names.
+---------------------
+P.S.
+---------------------
+1- This code does not consider the residue of molecules. So as a prerequisite,
+you may need to eliminate the non-target molecules after simulation and before
+runnig the code.
+2- The periodic boundary should be eliminated before running the code. We
+recommend doing -pbc whole and -pbc cluster using trjconv keyword (available)
+in GROMCAS or any similar procedure.
+3- If you have more than one type of molecule and you want to measure the
+aggregation without considering the type of molecule the code will works
+perfectly for "Closest" and "COM" Types while for the "Specific" the results
+will not be correct. The reasone is that you cannot tell atom_1 and atom_2 is
+related to which residue!
+4- If you have more than one type of molecule and you want to measure
+aggregation for each type of molecule separately, you can keep only the
+target molecule in the box and run the code multiple times for different
+molecule type.
+5- You will get an error if there is no aggregation in the box or if the
+aggregation distance choose too short.
+"""
+import numpy as np
+import pandas as pd
+import MDAnalysis as mda
+from MDAnalysis.lib import distances
+from MDAnalysis.analysis.base import (AnalysisBase)
+
+
+class Aggregation_size(AnalysisBase):  # subclass AnalysisBase
+    def __init__(self, atomgroup, number_of_molecules, Type="Closest",
+                 no_monomer=True, Gyr_calc=False, cut_off=False,
+                 atom_1=None, atom_2=None, **kwargs):
+        """
+        Set up the initial analysis parameters.
+        """
+        # must first run AnalysisBase.__init__ and pass the trajectory
+        trajectory = atomgroup.universe.trajectory
+        super(Aggregation_size, self).__init__(trajectory)
+        # set atomgroup as a property for access in other methods
+        self.atomgroup = atomgroup
+        self._molnum = number_of_molecules
+        self._type = Type
+        self._cut_off = cut_off
+        self._monomer = no_monomer
+        self._Gyr_calc = Gyr_calc
+        self._GYR = []
+
+    def closest_atom(self, d):
+        """
+        This function calculates the distance between the closest atom
+        of two molecules.
+        """
+        self._cut_off = self._cut_off or 3.5
+        for i in range(1, self._molnum+1):
+            ii = 0
+            for j in range(i+1, self._molnum+1):
+                residue1 = u.select_atoms(f"resid {i}").positions
+                residue2 = u.select_atoms(f"resid {j}").positions
+                dists = distances.distance_array(
+                    residue1, residue2, box=u.universe.dimensions,
+                    backend='OpenMP')
```

Comment:
Don't hard-code the backend, it might fail when not available. If you need this control, make `backend` a kwarg for init and then use `self.backend` here.